hoomd/operation.py

```diff
@@ -371,11 +241,14 @@ class _HOOMDBaseObject(_StatefulAttrBase, _DependencyRelation):
     detaching is removing an object from its simulation.
     """
     _reserved_default_attrs = {**_HOOMDGetSetAttrBase._reserved_default_attrs,
-                               '_cpp_obj': lambda: None,
-                               '_dependents': lambda: [],
-                               '_dependencies': lambda: []}
+                               '_cpp_obj': _ReturnCopies(None),
+                               '_dependents': _ReturnCopies([]),
```

Comment:
An alternative would be to define something like:

```python
class _Instantiator:
    def __init__(self, instance_class):
        self.instance_class = instance_class
    def __call__(self):
        return self.instance_class()

_dependents = _Instantiator(list)
```

Or even better yet, just provide a class that will create a new instance when called. 😉 
```
_dependents = list
```

Then you could just call the item to get a value if it's callable and return values like `None` directly.

```python
# Replace whatever calls _ReturnCopies with:
return value() if callable(value) else value
```

Reply:
Good thought, the last suggestion is way less over engineered.